src/NuGet.Core/NuGet.Protocol/Providers/RawSearchResourceV3Provider.cs

```diff
@@ -27,6 +29,15 @@ public RawSearchResourceV3Provider()
             {
                 var endpoints = serviceIndex.GetServiceEntryUris(ServiceTypes.SearchQueryService);
 
+                foreach (Uri endpoint in endpoints)
+                {
+                    PackageSource serviceSource = new PackageSource(endpoint.ToString(), endpoint.ToString());
```

Comment:
We need to minimize memory allocations to minimize performance slowdowns. Uri has a property `OriginalString`, which I expect can be used and shouldn't allocate.